src/main/java/com/cleanroommc/groovyscript/core/SideOnlyConfig.java

```diff
@@ -0,0 +1,120 @@
+package com.cleanroommc.groovyscript.core;
+
+import com.cleanroommc.groovyscript.GroovyScript;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.objects.ObjectOpenHashSet;
+import net.minecraftforge.fml.relauncher.FMLInjectionData;
+import net.minecraftforge.fml.relauncher.Side;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+public class SideOnlyConfig {
+
+    private static final MethodSet CLASS_MARKER = new MethodSet(true);
+    private static final Map<String, MethodSet> clientRemovals = new Object2ObjectOpenHashMap<>();
+    private static final Map<String, MethodSet> serverRemovals = new Object2ObjectOpenHashMap<>();
+
+    public static void clientOnly(String className, String member) {
+        serverRemovals.computeIfAbsent(className, k -> new MethodSet(false)).add(member);
+    }
+
+    public static void serverOnly(String className, String member) {
+        clientRemovals.computeIfAbsent(className, k -> new MethodSet(false)).add(member);
+    }
+
+    static void init() {
+        clientOnly("net.minecraftforge.common.config.Configuration", "setCategoryConfigEntryClass()");
+
+        initConfig((File) FMLInjectionData.data()[6]);
+    }
+
+    private static void initConfig(File minecraftHome) {
+        File scriptPath;
+        if (Boolean.parseBoolean(System.getProperty("groovyscript.use_examples_folder"))) {
+            scriptPath = new File(minecraftHome.getParentFile(), "examples");
+        } else {
+            scriptPath = new File(minecraftHome, "groovy");
+        }
+        File sideOnlyConfig = new File(scriptPath, "sideOnly.json");
```

Comment:
why doesnt `GroovyScript.scriptPath` work here?

Reply:
We cant guarantee that grs is already initialized here